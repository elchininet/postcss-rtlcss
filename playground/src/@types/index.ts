export type NumberOrString = string | number;
export type NumberOrStringArray = NumberOrString[];

export interface ObjectProps<T> {
    [key: string]: T;
}