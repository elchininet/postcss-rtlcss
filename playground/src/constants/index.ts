export enum COLORS {
    action = '#18AEC9',
    active = '#026C7F',
    gray = '#DDD',
    gray_dark = '#555',
    gray_darkest = '#1E1E1E',
    gray_light = '#EFEFEF',
    white = '#FFF'
}