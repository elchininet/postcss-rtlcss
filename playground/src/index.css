html,
body {
    height: 100%;
    margin: 0;
    padding: 0;
}

body {
    font-family: Arial, Helvetica, sans-serif;
    overscroll-behavior-y: contain;
}

#app {
    height: 100%;
}