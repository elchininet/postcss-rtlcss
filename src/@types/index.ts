import { Rule, AtRule } from 'postcss';

export enum Mode {
    combined = 'combined',
    override = 'override',
    diff = 'diff'
}

export enum Source {
    ltr = 'ltr',
    rtl = 'rtl'
}

export enum Autorename {
    disabled = 'disabled',
    flexible = 'flexible',
    strict = 'strict'
}

export type ModeValues = keyof typeof Mode;
export type SourceValues = keyof typeof Source;
export type AutorenameValues = keyof typeof Autorename;

export type strings = string | string[];

interface StringMapOptions {
    scope: '*' | 'selector' | 'url';
    ignoreCase: boolean;
}

export interface StringMap {
    name: string;
    priority: number;
    search: strings;
    replace: strings;
    options: StringMapOptions;
}

export interface PluginStringMap {
    name?: string;
    search: strings;
    replace: strings;
}

export interface PluginOptions {
    mode?: ModeValues;
    ltrPrefix?: strings;
    rtlPrefix?: strings;
    bothPrefix?: strings;
    safeBothPrefix?: boolean;
    ignorePrefixedRules?: boolean;
    source?: SourceValues;
    processUrls?: boolean;
    processKeyFrames?: boolean;
    processEnv?: boolean;
    useCalc?: boolean;
    stringMap?: PluginStringMap[];
    autoRename?: AutorenameValues;
    greedy?: boolean;
    aliases?: Record<string, string>;
}

export interface PluginOptionsNormalized extends Omit<Required<PluginOptions>, 'stringMap'> {
    stringMap: StringMap[];
}

export interface RulesObject {
    rule: Rule;
    ruleLTR: Rule;
    ruleRTL: Rule;
    ruleBoth: Rule;
    ruleSafe: Rule;
}

export interface AtRulesObject {
    atRuleParams: string;
    atRule: AtRule;
    atRuleFlipped: AtRule;
}

export interface AtRulesStringMapObject {
    name: string;
    nameFlipped: string;
}

export interface AtRulesStringMap {
    [key: string]: AtRulesStringMapObject;
}

export interface DeclarationsData {
    [key: string]: {
        overridden: string | null;
        overrides: string[];
    };
}

export interface ControlDirective {
    block?: string;
    directive: string;
    option?: string;
}