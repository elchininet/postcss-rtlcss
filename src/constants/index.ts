export const COMMENT_TYPE = 'comment';
export const DECLARATION_TYPE = 'decl';
export const RULE_TYPE = 'rule';
export const AT_RULE_TYPE = 'atrule';
export const STRING_TYPE = 'string';
export const BOOLEAN_TYPE = 'boolean';
export const KEYFRAMES_NAME = 'keyframes';
export const ANIMATION_PROP = 'animation';
export const ANIMATION_NAME_PROP = 'animation-name';
export const RTL_COMMENT_REGEXP = /rtl:/;
export const RTL_CONTROL_DIRECTIVE_REG_EXP = /^\/\*!? *rtl:?(\w+)?:(\w+):?([^*]*?) *\*\/$/;
export const FLIP_PROPERTY_REGEXP = /(right|left)/i;

export enum CONTROL_DIRECTIVE {
    IGNORE = 'ignore'
}

export enum CONTROL_DIRECTIVE_BLOCK {
    BEGIN = 'begin',
    END = 'end'
}