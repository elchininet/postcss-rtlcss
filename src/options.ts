import { Mode, Source } from '@types';

export { Mode, Source };