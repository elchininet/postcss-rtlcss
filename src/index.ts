import postcss, { Root, Transformer } from 'postcss';
import { PluginOptions, RulesObject } from '@types';
import { parseOptions } from '@utilities/options';
import { parseRules } from '@rules/parser';
import { appendRules } from '@utilities/rules';
export { PluginOptions, Mode, Source, PluginStringMap,  } from '@types';

const transformer = (options: PluginOptions = {}): Transformer  => (
    (css: Root): void => {
        const optionsParsed = parseOptions(options);
        const result: RulesObject[] = parseRules(css, optionsParsed);
        appendRules(result);           
    }
);

export const postcssRTLCSS = postcss.plugin<PluginOptions>('postcss-rtlcss', transformer);