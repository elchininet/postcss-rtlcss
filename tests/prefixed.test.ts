import postcss from 'postcss';
import postcssRTLCSS from '../src';
import { PluginOptions } from '../src/@types';
import {
    readCSSFile,
    runTests,
    createSnapshotFileName
} from './utils';
import 'jest-specific-snapshot';

const BASE_NAME = 'prefixed';

runTests({ processUrls: true }, (pluginOptions: PluginOptions): void => {

    describe(`[[Mode: ${pluginOptions.mode}]] Prefixed Tests: `, (): void => {

        let input = '';

        beforeEach(async (): Promise<void> => {
            input = input || await readCSSFile('input-prefixed.css');
        });

        it('Prefixed default', (): void => {
            const options: PluginOptions = { ...pluginOptions };
            const output = postcss([postcssRTLCSS(options)]).process(input);
            expect(output.css).toMatchSpecificSnapshot(
                createSnapshotFileName(BASE_NAME,'prefixed-default', pluginOptions.mode)
            );
            expect(output.warnings()).toHaveLength(0);
        });

        it('Custom prefixes', (): void => {
            const options: PluginOptions = { ...pluginOptions, rtlPrefix: '.rtl', ltrPrefix: '.ltr' };
            const output = postcss([postcssRTLCSS(options)]).process(input);
            expect(output.css).toMatchSpecificSnapshot(
                createSnapshotFileName(BASE_NAME,'custom-prefixes', pluginOptions.mode)
            );
            expect(output.warnings()).toHaveLength(0);
        });

        it('Prefixed with array', (): void => {
            const options: PluginOptions = { ...pluginOptions, rtlPrefix: ['.rtl', '[dir="rtl"]'], ltrPrefix: ['.ltr', '[dir="ltr"]'] };
            const output = postcss([postcssRTLCSS(options)]).process(input);
            expect(output.css).toMatchSpecificSnapshot(
                createSnapshotFileName(BASE_NAME,'prefixed-with-array', pluginOptions.mode)
            );
            expect(output.warnings()).toHaveLength(0);
        });

        it('ignorePrefixedRules false', (): void => {
            const options: PluginOptions = { ...pluginOptions, ignorePrefixedRules: false };
            const output = postcss([postcssRTLCSS(options)]).process(input);
            expect(output.css).toMatchSpecificSnapshot(
                createSnapshotFileName(BASE_NAME,'ignore-prefixed-rules-false', pluginOptions.mode)
            );
            expect(output.warnings()).toHaveLength(0);
        });

    });

});