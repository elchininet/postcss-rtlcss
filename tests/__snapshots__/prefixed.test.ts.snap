// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[[Mode: combined]] Prefixed Tests:  Custom prefixes 1`] = `
".test {
    color: red;
}

.rtl .test {
    left: 10px;
}

.ltr .test {
    right: 10px;
}

.ltr .rtlignore .test2 {
    padding: 2px 4px 8px 16px;
}

.rtl .rtlignore .test2 {
    padding: 2px 16px 8px 4px;
}

.ltr .ltrignore .test2 {
    margin: 2px 4px 8px 16px;
}

.rtl .ltrignore .test2 {
    margin: 2px 16px 8px 4px;
}

.ltr [dir=\\"ltr\\"] .test3 {
    text-align: left;
}

.rtl [dir=\\"ltr\\"] .test3 {
    text-align: right;
}

.ltr [dir=\\"rtl\\"] .test3 {
    text-align: right;
}

.rtl [dir=\\"rtl\\"] .test3 {
    text-align: left;
}

.ltr .test4 {
    color: #EFEFEF;
    left: 10px;
    width: 100%;    
}

.rtl [dir=\\"rtl\\"] .test5 {
    transform: translate(10px, 20px);
}

.rtl .test6 {
    left: 5px;
}

.ltr .test7-ltr {
    background-image: url(\\"/icons/icon-left.png\\");
}

.rtl .test7-rtl {
    background-image: url(\\"/icons/icon-right.png\\");
}

.ltr [dir=\\"ltr\\"] .test8-ltr {
    background-image: url(\\"/icons/icon-ltr.png\\");
}

.rtl [dir=\\"ltr\\"] .test8-ltr {
    background-image: url(\\"/icons/icon-rtl.png\\");
}

.ltr [dir=\\"rtl\\"] .test8-rtl {
    background-image: url(\\"/icons/icon-rtl.png\\");
}

.rtl [dir=\\"rtl\\"] .test8-rtl {
    background-image: url(\\"/icons/icon-ltr.png\\");
}"
`;

exports[`[[Mode: combined]] Prefixed Tests:  Prefixed default 1`] = `
".test {
    color: red;
}

[dir=\\"ltr\\"] .rtl .test {
    left: 10px;
}

[dir=\\"rtl\\"] .rtl .test {
    right: 10px;
}

[dir=\\"ltr\\"] .ltr .test {
    right: 10px;
}

[dir=\\"rtl\\"] .ltr .test {
    left: 10px;
}

[dir=\\"ltr\\"] .rtlignore .test2 {
    padding: 2px 4px 8px 16px;
}

[dir=\\"rtl\\"] .rtlignore .test2 {
    padding: 2px 16px 8px 4px;
}

[dir=\\"ltr\\"] .ltrignore .test2 {
    margin: 2px 4px 8px 16px;
}

[dir=\\"rtl\\"] .ltrignore .test2 {
    margin: 2px 16px 8px 4px;
}

[dir=\\"ltr\\"] .test3 {
    text-align: left;
}

[dir=\\"rtl\\"] .test3 {
    text-align: right;
}

[dir=\\"rtl\\"] .ltr .test4 {
    color: #EFEFEF;
    left: 10px;
    width: 100%;    
}

[dir=\\"rtl\\"] .test5 {
    transform: translate(10px, 20px);
}

[dir=\\"rtl\\"] .test6 {
    left: 5px;
}

[dir=\\"ltr\\"] .ltr .test7-ltr {
    background-image: url(\\"/icons/icon-left.png\\");
}

[dir=\\"rtl\\"] .ltr .test7-ltr {
    background-image: url(\\"/icons/icon-right.png\\");
}

[dir=\\"ltr\\"] .rtl .test7-rtl {
    background-image: url(\\"/icons/icon-right.png\\");
}

[dir=\\"rtl\\"] .rtl .test7-rtl {
    background-image: url(\\"/icons/icon-left.png\\");
}

[dir=\\"ltr\\"] .test8-ltr {
    background-image: url(\\"/icons/icon-ltr.png\\");
}

[dir=\\"rtl\\"] .test8-rtl {
    background-image: url(\\"/icons/icon-rtl.png\\");
}"
`;

exports[`[[Mode: combined]] Prefixed Tests:  Prefixed with array 1`] = `
".test {
    color: red;
}

.rtl .test {
    left: 10px;
}

.ltr .test {
    right: 10px;
}

.ltr .rtlignore .test2, [dir=\\"ltr\\"] .rtlignore .test2 {
    padding: 2px 4px 8px 16px;
}

.rtl .rtlignore .test2, [dir=\\"rtl\\"] .rtlignore .test2 {
    padding: 2px 16px 8px 4px;
}

.ltr .ltrignore .test2, [dir=\\"ltr\\"] .ltrignore .test2 {
    margin: 2px 4px 8px 16px;
}

.rtl .ltrignore .test2, [dir=\\"rtl\\"] .ltrignore .test2 {
    margin: 2px 16px 8px 4px;
}

[dir=\\"ltr\\"] .test3 {
    text-align: left;
}

[dir=\\"rtl\\"] .test3 {
    text-align: right;
}

.ltr .test4 {
    color: #EFEFEF;
    left: 10px;
    width: 100%;    
}

[dir=\\"rtl\\"] .test5 {
    transform: translate(10px, 20px);
}

.rtl .test6, [dir=\\"rtl\\"] .test6 {
    left: 5px;
}

.ltr .test7-ltr {
    background-image: url(\\"/icons/icon-left.png\\");
}

.rtl .test7-rtl {
    background-image: url(\\"/icons/icon-right.png\\");
}

[dir=\\"ltr\\"] .test8-ltr {
    background-image: url(\\"/icons/icon-ltr.png\\");
}

[dir=\\"rtl\\"] .test8-rtl {
    background-image: url(\\"/icons/icon-rtl.png\\");
}"
`;

exports[`[[Mode: combined]] Prefixed Tests:  ignorePrefixedRules false 1`] = `
".test {
    color: red;
}

[dir=\\"ltr\\"] .rtl .test {
    left: 10px;
}

[dir=\\"rtl\\"] .rtl .test {
    right: 10px;
}

[dir=\\"ltr\\"] .ltr .test {
    right: 10px;
}

[dir=\\"rtl\\"] .ltr .test {
    left: 10px;
}

[dir=\\"ltr\\"] .rtlignore .test2 {
    padding: 2px 4px 8px 16px;
}

[dir=\\"rtl\\"] .rtlignore .test2 {
    padding: 2px 16px 8px 4px;
}

[dir=\\"ltr\\"] .ltrignore .test2 {
    margin: 2px 4px 8px 16px;
}

[dir=\\"rtl\\"] .ltrignore .test2 {
    margin: 2px 16px 8px 4px;
}

[dir=\\"ltr\\"] [dir=\\"ltr\\"] .test3 {
    text-align: left;
}

[dir=\\"rtl\\"] [dir=\\"ltr\\"] .test3 {
    text-align: right;
}

[dir=\\"ltr\\"] [dir=\\"rtl\\"] .test3 {
    text-align: right;
}

[dir=\\"rtl\\"] [dir=\\"rtl\\"] .test3 {
    text-align: left;
}

[dir=\\"rtl\\"] .ltr .test4 {
    color: #EFEFEF;
    left: 10px;
    width: 100%;    
}

[dir=\\"rtl\\"] [dir=\\"rtl\\"] .test5 {
    transform: translate(10px, 20px);
}

[dir=\\"rtl\\"] .test6 {
    left: 5px;
}

[dir=\\"ltr\\"] .ltr .test7-ltr {
    background-image: url(\\"/icons/icon-left.png\\");
}

[dir=\\"rtl\\"] .ltr .test7-ltr {
    background-image: url(\\"/icons/icon-right.png\\");
}

[dir=\\"ltr\\"] .rtl .test7-rtl {
    background-image: url(\\"/icons/icon-right.png\\");
}

[dir=\\"rtl\\"] .rtl .test7-rtl {
    background-image: url(\\"/icons/icon-left.png\\");
}

[dir=\\"ltr\\"] [dir=\\"ltr\\"] .test8-ltr {
    background-image: url(\\"/icons/icon-ltr.png\\");
}

[dir=\\"rtl\\"] [dir=\\"ltr\\"] .test8-ltr {
    background-image: url(\\"/icons/icon-rtl.png\\");
}

[dir=\\"ltr\\"] [dir=\\"rtl\\"] .test8-rtl {
    background-image: url(\\"/icons/icon-rtl.png\\");
}

[dir=\\"rtl\\"] [dir=\\"rtl\\"] .test8-rtl {
    background-image: url(\\"/icons/icon-ltr.png\\");
}"
`;

exports[`[[Mode: diff]] Prefixed Tests:  Custom prefixes 1`] = `
"

.rtlignore .test2 {
    padding: 2px 16px 8px 4px;
}

.ltrignore .test2 {
    margin: 2px 16px 8px 4px;
}

[dir=\\"ltr\\"] .test3 {
    text-align: right;
}

[dir=\\"rtl\\"] .test3 {
    text-align: left;
}

.ltr .test4 {
    color: #EFEFEF;
    left: 10px;
    width: 100%;    
}

[dir=\\"rtl\\"] .test5 {
    transform: translate(10px, 20px);
}

.test6 {
    left: 5px;
}

[dir=\\"ltr\\"] .test8-ltr {
    background-image: url(\\"/icons/icon-rtl.png\\");
}

[dir=\\"rtl\\"] .test8-rtl {
    background-image: url(\\"/icons/icon-ltr.png\\");
}"
`;

exports[`[[Mode: diff]] Prefixed Tests:  Prefixed default 1`] = `
"

.rtl .test {
    left: auto;
    right: 10px;
}

.ltr .test {
    right: auto;
    left: 10px;
}

.rtlignore .test2 {
    padding: 2px 16px 8px 4px;
}

.ltrignore .test2 {
    margin: 2px 16px 8px 4px;
}

.ltr .test4 {
    color: #EFEFEF;
    left: 10px;
    width: 100%;    
}

[dir=\\"rtl\\"] .test5 {
    transform: translate(10px, 20px);
}

.test6 {
    left: 5px;
}

.ltr .test7-ltr {
    background-image: url(\\"/icons/icon-right.png\\");
}

.rtl .test7-rtl {
    background-image: url(\\"/icons/icon-left.png\\");
}"
`;

exports[`[[Mode: diff]] Prefixed Tests:  Prefixed with array 1`] = `
"

.rtlignore .test2 {
    padding: 2px 16px 8px 4px;
}

.ltrignore .test2 {
    margin: 2px 16px 8px 4px;
}

.ltr .test4 {
    color: #EFEFEF;
    left: 10px;
    width: 100%;    
}

[dir=\\"rtl\\"] .test5 {
    transform: translate(10px, 20px);
}

.test6 {
    left: 5px;
}"
`;

exports[`[[Mode: diff]] Prefixed Tests:  ignorePrefixedRules false 1`] = `
"

.rtl .test {
    left: auto;
    right: 10px;
}

.ltr .test {
    right: auto;
    left: 10px;
}

.rtlignore .test2 {
    padding: 2px 16px 8px 4px;
}

.ltrignore .test2 {
    margin: 2px 16px 8px 4px;
}

[dir=\\"ltr\\"] .test3 {
    text-align: right;
}

[dir=\\"rtl\\"] .test3 {
    text-align: left;
}

.ltr .test4 {
    color: #EFEFEF;
    left: 10px;
    width: 100%;    
}

[dir=\\"rtl\\"] .test5 {
    transform: translate(10px, 20px);
}

.test6 {
    left: 5px;
}

.ltr .test7-ltr {
    background-image: url(\\"/icons/icon-right.png\\");
}

.rtl .test7-rtl {
    background-image: url(\\"/icons/icon-left.png\\");
}

[dir=\\"ltr\\"] .test8-ltr {
    background-image: url(\\"/icons/icon-rtl.png\\");
}

[dir=\\"rtl\\"] .test8-rtl {
    background-image: url(\\"/icons/icon-ltr.png\\");
}"
`;

exports[`[[Mode: override]] Prefixed Tests:  Custom prefixes 1`] = `
".test {
    color: red;
}

.rtl .test {
    left: 10px;
}

.ltr .test {
    right: 10px;
}

.rtlignore .test2 {
    padding: 2px 4px 8px 16px;
}

.rtl .rtlignore .test2 {
    padding: 2px 16px 8px 4px;
}

.ltrignore .test2 {
    margin: 2px 4px 8px 16px;
}

.rtl .ltrignore .test2 {
    margin: 2px 16px 8px 4px;
}

[dir=\\"ltr\\"] .test3 {
    text-align: left;
}

.rtl [dir=\\"ltr\\"] .test3 {
    text-align: right;
}

[dir=\\"rtl\\"] .test3 {
    text-align: right;
}

.rtl [dir=\\"rtl\\"] .test3 {
    text-align: left;
}

.ltr .test4 {
    color: #EFEFEF;
    left: 10px;
    width: 100%;    
}

.rtl [dir=\\"rtl\\"] .test5 {
    transform: translate(10px, 20px);
}

.rtl .test6 {
    left: 5px;
}

.ltr .test7-ltr {
    background-image: url(\\"/icons/icon-left.png\\");
}

.rtl .test7-rtl {
    background-image: url(\\"/icons/icon-right.png\\");
}

[dir=\\"rtl\\"] .test8-rtl {
    background-image: url(\\"/icons/icon-ltr.png\\");
}

.rtl [dir=\\"ltr\\"] .test8-ltr {
    background-image: url(\\"/icons/icon-rtl.png\\");
}

[dir=\\"ltr\\"] .test8-ltr {
    background-image: url(\\"/icons/icon-rtl.png\\");
}

.rtl [dir=\\"rtl\\"] .test8-rtl {
    background-image: url(\\"/icons/icon-ltr.png\\");
}"
`;

exports[`[[Mode: override]] Prefixed Tests:  Prefixed default 1`] = `
".test {
    color: red;
}

.rtl .test {
    left: 10px;
}

[dir=\\"rtl\\"] .rtl .test {
    left: auto;
    right: 10px;
}

.ltr .test {
    right: 10px;
}

[dir=\\"rtl\\"] .ltr .test {
    right: auto;
    left: 10px;
}

.rtlignore .test2 {
    padding: 2px 4px 8px 16px;
}

[dir=\\"rtl\\"] .rtlignore .test2 {
    padding: 2px 16px 8px 4px;
}

.ltrignore .test2 {
    margin: 2px 4px 8px 16px;
}

[dir=\\"rtl\\"] .ltrignore .test2 {
    margin: 2px 16px 8px 4px;
}

[dir=\\"ltr\\"] .test3 {
    text-align: left;
}

[dir=\\"rtl\\"] .test3 {
    text-align: right;
}

[dir=\\"rtl\\"] .ltr .test4 {
    color: #EFEFEF;
    left: 10px;
    width: 100%;    
}

[dir=\\"rtl\\"] .test5 {
    transform: translate(10px, 20px);
}

[dir=\\"rtl\\"] .test6 {
    left: 5px;
}

.rtl .test7-rtl {
    background-image: url(\\"/icons/icon-left.png\\");
}

[dir=\\"rtl\\"] .ltr .test7-ltr {
    background-image: url(\\"/icons/icon-right.png\\");
}

.ltr .test7-ltr {
    background-image: url(\\"/icons/icon-right.png\\");
}

[dir=\\"rtl\\"] .rtl .test7-rtl {
    background-image: url(\\"/icons/icon-left.png\\");
}

[dir=\\"ltr\\"] .test8-ltr {
    background-image: url(\\"/icons/icon-ltr.png\\");
}

[dir=\\"rtl\\"] .test8-rtl {
    background-image: url(\\"/icons/icon-rtl.png\\");
}"
`;

exports[`[[Mode: override]] Prefixed Tests:  Prefixed with array 1`] = `
".test {
    color: red;
}

.rtl .test {
    left: 10px;
}

.ltr .test {
    right: 10px;
}

.rtlignore .test2 {
    padding: 2px 4px 8px 16px;
}

.rtl .rtlignore .test2, [dir=\\"rtl\\"] .rtlignore .test2 {
    padding: 2px 16px 8px 4px;
}

.ltrignore .test2 {
    margin: 2px 4px 8px 16px;
}

.rtl .ltrignore .test2, [dir=\\"rtl\\"] .ltrignore .test2 {
    margin: 2px 16px 8px 4px;
}

[dir=\\"ltr\\"] .test3 {
    text-align: left;
}

[dir=\\"rtl\\"] .test3 {
    text-align: right;
}

.ltr .test4 {
    color: #EFEFEF;
    left: 10px;
    width: 100%;    
}

[dir=\\"rtl\\"] .test5 {
    transform: translate(10px, 20px);
}

.rtl .test6, [dir=\\"rtl\\"] .test6 {
    left: 5px;
}

.ltr .test7-ltr {
    background-image: url(\\"/icons/icon-left.png\\");
}

.rtl .test7-rtl {
    background-image: url(\\"/icons/icon-right.png\\");
}

[dir=\\"ltr\\"] .test8-ltr {
    background-image: url(\\"/icons/icon-ltr.png\\");
}

[dir=\\"rtl\\"] .test8-rtl {
    background-image: url(\\"/icons/icon-rtl.png\\");
}"
`;

exports[`[[Mode: override]] Prefixed Tests:  ignorePrefixedRules false 1`] = `
".test {
    color: red;
}

.rtl .test {
    left: 10px;
}

[dir=\\"rtl\\"] .rtl .test {
    left: auto;
    right: 10px;
}

.ltr .test {
    right: 10px;
}

[dir=\\"rtl\\"] .ltr .test {
    right: auto;
    left: 10px;
}

.rtlignore .test2 {
    padding: 2px 4px 8px 16px;
}

[dir=\\"rtl\\"] .rtlignore .test2 {
    padding: 2px 16px 8px 4px;
}

.ltrignore .test2 {
    margin: 2px 4px 8px 16px;
}

[dir=\\"rtl\\"] .ltrignore .test2 {
    margin: 2px 16px 8px 4px;
}

[dir=\\"ltr\\"] .test3 {
    text-align: left;
}

[dir=\\"rtl\\"] [dir=\\"ltr\\"] .test3 {
    text-align: right;
}

[dir=\\"rtl\\"] .test3 {
    text-align: right;
}

[dir=\\"rtl\\"] [dir=\\"rtl\\"] .test3 {
    text-align: left;
}

[dir=\\"rtl\\"] .ltr .test4 {
    color: #EFEFEF;
    left: 10px;
    width: 100%;    
}

[dir=\\"rtl\\"] [dir=\\"rtl\\"] .test5 {
    transform: translate(10px, 20px);
}

[dir=\\"rtl\\"] .test6 {
    left: 5px;
}

.rtl .test7-rtl {
    background-image: url(\\"/icons/icon-left.png\\");
}

[dir=\\"rtl\\"] .ltr .test7-ltr {
    background-image: url(\\"/icons/icon-right.png\\");
}

.ltr .test7-ltr {
    background-image: url(\\"/icons/icon-right.png\\");
}

[dir=\\"rtl\\"] .rtl .test7-rtl {
    background-image: url(\\"/icons/icon-left.png\\");
}

[dir=\\"rtl\\"] .test8-rtl {
    background-image: url(\\"/icons/icon-ltr.png\\");
}

[dir=\\"rtl\\"] [dir=\\"ltr\\"] .test8-ltr {
    background-image: url(\\"/icons/icon-rtl.png\\");
}

[dir=\\"ltr\\"] .test8-ltr {
    background-image: url(\\"/icons/icon-rtl.png\\");
}

[dir=\\"rtl\\"] [dir=\\"rtl\\"] .test8-rtl {
    background-image: url(\\"/icons/icon-ltr.png\\");
}"
`;
